### PR TITLE
Install Extension feature - Azure App Service Manage Task

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureAppServiceManage/azureappservicemanage.ts [hunk 1]
```diff
@@ -40,6 +41,7 @@ async function run() {
         var swapWithProduction = tl.getBoolInput('SwapWithProduction', false);
         var targetSlot: string = tl.getInput('TargetSlot', false);
         var preserveVnet: boolean = tl.getBoolInput('PreserveVnet', false);
+        var extensionList = tl.getInput('ExtensionsList', false);
```
Comment (kmkumaran) [inline Tasks/AzureAppServiceManage/azureappservicemanage.ts]: To confirm - This input will have the list of extension Id is it? If so, if you plan to support dynamic pick list for this input, will it still remain as Id?

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 2]
```diff
@@ -416,16 +416,18 @@ export async function startAppService(endpoint, resourceGroupName: string, webAp
         'Authorization': 'Bearer '+ accessToken
     };
     var webAppNameWithSlot = (specifySlotFlag) ? webAppName + '-' + slotName : webAppName;
+    tl.debug('Request to start App Service: ' + url);
```
Comment (kmkumaran) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: What are these changes in start app service? They don't seem related to the extension changes.

File: Tasks/AzureAppServiceManage/extensionmanage.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import Q = require('q');
```
Comment (kmkumaran) [inline Tasks/AzureAppServiceManage/extensionmanage.ts]: Please add L0

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 2 above)
Comment (vincentdass) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Yes, they are not related to extensions. just adding debug logs.

File: Tasks/AzureAppServiceManage/azureappservicemanage.ts (hunk 1 above)
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/azureappservicemanage.ts]: yes, Input will be a  group of extension ID.

File: Tasks/AzureAppServiceManage/azureappservicemanage.ts [hunk 4]
```diff
@@ -66,6 +68,16 @@ async function run() {
                 console.log(await azureRmUtil.stopAppService(endPoint, resourceGroupName, webAppName, specifySlotFlag, slotName));
                 break;
             }
+            case "Install Extensions": {
+                resourceGroupName = (specifySlotFlag ? resourceGroupName : await azureRmUtil.getResourceGroupName(endPoint, webAppName));
+                var publishingProfile = await azureRmUtil.getAzureRMWebAppPublishProfile(endPoint, webAppName, resourceGroupName, specifySlotFlag, slotName);
+                tl.debug('Retrieved publishing Profile');
+                var anyExtensionInstalled = await extensionManage.installExtensions(publishingProfile, extensionList.split(','));
+                if(!anyExtensionInstalled) {
+                    tl.debug('No new extension installed. Skipping Restart App Service.');
```
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/azureappservicemanage.ts]: We restart the app service, only when extensions are installed.
If extension is already installed, then we skip the installation and no restart done.

File: Tasks/AzureAppServiceManage/extensionmanage.ts (hunk 3 above)
Comment (vincentdass) [inline Tasks/AzureAppServiceManage/extensionmanage.ts]: done
